mlstream/datautils.py

```diff
@@ -108,43 +110,40 @@ def load_forcings_lumped(data_root: Path, basins: List = None, file_format: str
         which contains one .rvt/.csv/.txt -file per basin.
     basins : List, optional
         List of basins for which to return data. Default (None) returns data for all basins.
-    file_format : str, optional
-        Format of the forcing files. Default format is 'rvt' but 'csv' is also supported.
-        - rvt format: fourth line contains column names (",\s+"-sepatated), following lines contain ",\s*"-separated
-            data. Second line starts with start date (yyyy-mm-dd). Last line is footer.
-        - txt/csv format: first line is header, following lines are comma-separated. First column contains dates
-            as yyyy-mm-dd.
 
     Returns
     -------
     dict
         Dictionary of forcings (pd.DataFrame) per basin
     """
-    if file_format not in ['rvt', 'csv', 'txt']:
-        raise NotImplementedError(f"Forcing format {file_format} not supported.")
-
     lumped_dir = data_root / 'forcings' / 'lumped'
-    basin_files = lumped_dir.glob('*.' + file_format)
+    basin_files = lumped_dir.glob('*')

```

Comment:
This will match directories, but should only match files. 
Also, I think it's better not to match file extensions we don't support (i.e., anything other than rvt/csv/txt), so people can have e.g. a readme.md in the directory without problems.
Maybe print a warning if there are unsupported file endings, and a different warning if a file with correct ending couldn't be read.

Reply:
I pushed changes for reading only files from the directory.
If the file extension is not supported then already it will throw a `NotImplementedError` exception. There is an if-elif-else ladder to handle each supported extension appropriately, and at the end of the ladder, `NotImplementedError` is thrown. Therefore, the warning for an unsupported file type and inability to read the file will be different.